src/Avalonia.X11/NativeDialogs/Gtk.cs

```diff
@@ -284,35 +180,59 @@ private static void GtkThread(TaskCompletionSource<bool> tcs)
 
                 if (!gtk_init_check(0, IntPtr.Zero))
                 {
-                    tcs.SetResult(false);
-                    return;
+                    return false;
                 }
 
                 IntPtr app;
                 using (var utf = new Utf8Buffer($"avalonia.app.a{Guid.NewGuid():N}"))
                     app = gtk_application_new(utf, 0);
```

Comment:
A question. assuming, these APIs will be used from the GTK.
Will `GTK Application -> Avalonia App -> Dialog specific GTK Application` work well? Don't we need to check for existing gtk app?  It's probably out of scope of this PR either way.

Reply:
I'm not really sure if it's even required to call. I suspect that gtk_init_check should be sufficient.